pcf/particle/aws/ecr/ecr_repository.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pcf.core import State
+from pcf.util import pcf_util
+from pcf.core.aws_resource import AWSResource
+from pcf.core.pcf_exceptions import *
+import logging
+
+logger = logging.getLogger(__name__)
+
+class ECRRepository(AWSResource):
+    """
+    This is the implementation of Amazon's EC2 Container Registry (ECR) Repository.
+    """
+
+    flavor = "ecr_repository"
+
+    state_lookup = {
+        "available": State.running,
+        "pending": State.pending,
+        "missing": State.terminated
+    }
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0
+    }
+
+    DEFINITION_FILTER = {
+        'repositoryName',
+        'tags'
+    }
+
+    RETURN_PARAM_FILTER = {
+    }
+
+    UNIQUE_KEYS = ["aws_resource.repositoryName"]
+
+    def __init__(self, particle_definition, session=None):
+        super().__init__(particle_definition, "ecr", session=session)
+        self._set_unique_keys()
+        self.repositoryName = self.desired_state_definition.get("repositoryName")
+
+    def _set_unique_keys(self):
+        """
+        Logic that sets keys from state definition that are used to uniquely identify the ECR Repository
+
+        """
+        self.unique_keys = ECRRepository.UNIQUE_KEYS
+
+    def _terminate(self):
+        """
+        Calls boto3 delete_repository()
+
+        Returns:
+            boto3 delete_repository() response
+        """
+        resp = self.client.delete_repository(repositoryName =self.repositoryName)
+        return resp
+
+    def _start(self):
+        """
+        Starts the events particle that matches desired state definition
+
+        Returns:
+            response of boto3 create_repository()
+        """
+        start_definition = pcf_util.param_filter(self.get_desired_state_definition(), ECRRepository.DEFINITION_FILTER)
+        return self.client.create_repository(**start_definition)
```

Comment:
so you can create with tags, but the describe doesnt return tags?

Reply:
Correct :) Super strange.